crates/solidity-v2/outputs/cargo/parser/src/lexer/temp_sourcify/mod.rs

```diff
@@ -0,0 +1,129 @@
+
+        while let Some(lexeme) = instance.lexer.next_lexeme() {
+            instance.add_error(format!(
+                "Lexer v2 produced an extra lexeme at the end of input: {lexeme:?}",
+            ));
+        }
+
+        instance.errors
+    }
+
+    fn check_node(&mut self) {
+        match self.cursor.node() {
+            Node::Nonterminal(nonterminal) => {
+                // TODO(v2): current grammar definition has a bug, where it doesn't include all referenced terminals
+                // in every lexical context (e.g. allows Yul nonterminals to reference Solidity terminals, like `StringLiteral`).
+                // This is to be fixed in the next PR, but until then, let's "simulate" the switch,
+                // and skip the nodes that switch contexts for now:
+                match nonterminal.kind {
+                    NonterminalKind::Pragma => {
+                        self.lexer.switch_context(ContextKind::Pragma);
+                        self.lexer.bump(self.cursor.text_range().utf8().len());
+                        self.lexer.switch_context(ContextKind::Solidity);
+                        return;
+                    }
+                    NonterminalKind::AssemblyStatement => {
+                        self.lexer.switch_context(ContextKind::Yul);
+                        self.lexer.bump(self.cursor.text_range().utf8().len());
+                        self.lexer.switch_context(ContextKind::Solidity);
+                        return;
+                    }
+                    _ => {}
+                }
+
+                if self.cursor.go_to_first_child() {
+                    self.check_node();
+
+                    while self.cursor.go_to_next_sibling() {
+                        self.check_node();
+                    }
+
+                    self.cursor.go_to_parent();
+                }
+            }
+
+            Node::Terminal(terminal) => match self.lexer.next_lexeme() {
+                None => {
+                    self.add_error(
+                        "V2 lexer stopped producing lexemes, but v1 tree still has more terminals"
+                            .to_string(),
+                    );
+                }
+                Some(lexeme) => {
+                    let v1_range = self.cursor.text_range().utf8();
+                    let v2_range = lexeme.range;
+
+                    let v1_kind = terminal.kind.as_ref();
+                    let v2_kind: &'static str = lexeme.kind.into();
+
+                    let v1_output = format!("{v1_kind:?} @ {v1_range:?}");
+                    let v2_output = format!("{v2_kind:?} @ {v2_range:?}");
+
+                    if !match (v1_kind, v2_kind) {
+                        ("Identifier", v2_kind) if v2_kind.ends_with("_Unreserved") => true,
+                        (v1_kind, v2_kind) if format!("{v1_kind}_Reserved") == v2_kind => true,
+                        (v1_kind, v2_kind) if format!("{v1_kind}_Unreserved") == v2_kind => true,
```

Comment:
Out of curiosity, do you have an example in mind that this case is true?

My understanding was that if the new lexer produces `UsingKeyword_Unreserved` (for example) this should be `Identifier` in the old lexer, and not `UsingKeyword`

Reply:
Right, I still think `finney` could be an identifier, but because it's also `enabled = Till("0.7.0")`. But there are other ones that are enabled but not reserved at the same time (like `abicoder`) that you really don't want to lex as an identifier, since I guess wether it's an identifier or a keyword depends on the context (resolved during parsing).